homeassistant/components/sentry/__init__.py

```diff
@@ -42,16 +99,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     hass.data[DOMAIN] = conf
 
-    # https://docs.sentry.io/platforms/python/logging/
-    sentry_logging = LoggingIntegration(
-        level=logging.INFO,  # Capture info and above as breadcrumbs
-        event_level=logging.ERROR,  # Send errors as events
-    )
+    with_tracing = conf.get(CONF_TRACING)
 
     sentry_sdk.init(
         dsn=conf.get(CONF_DSN),
         environment=conf.get(CONF_ENVIRONMENT),
-        integrations=[sentry_logging],
+        integrations=[
+            # https://docs.sentry.io/platforms/python/logging/
+            LoggingIntegration(
+                level=logging.INFO,  # Capture info and above as breadcrumbs
+                event_level=logging.ERROR,  # Send errors as events
+            ),
+            *_get_integrations(),
+        ],
+        traces_sample_rate=(
+            conf.get(CONF_TRACING_SAMPLE_RATE, 1.0) or 0.0 if with_tracing else 0.0
```

Comment:
Please put the default in the config schema instead.

Why do we need this part?
`or 0.0 if with_tracing else 0.0`